File: codewars/python/kyu6/digital_root.py
# https://www.codewars.com/kata/541c8630095125aba6000c00/train/python

def num_sum(n): 
	s = 0

	while n != 0: 
		s += n % 10
		n //= 10

	return s

def digital_root(n):

	temp = n
	while n // 10 != 0: 
		
		n = num_sum(n)

	return n



assert digital_root(16) == 7
assert digital_root(942) == 6
assert digital_root(132189) == 6
assert digital_root(493193) == 2

File: c/6kyu/digital_root.c
// https://www.codewars.com/kata/541c8630095125aba6000c00
#include "../supporting_functions/write_num.c"

int digit_sum(int num)
{
	int sum;

	sum = 0;
	while (num)
	{
		sum += num % 10;
		num /= 10;
	}
	return (sum);
}

int digital_root(int n) {

	while (n / 10 != 0)
		n = digit_sum(n);
	return (n);
}

int main(int argc, char const *argv[])
{
	write_num(digital_root(132189));
	return 0;
}
